app/controllers/fhir/r4/api_controller.rb

```diff
@@ -111,7 +121,7 @@ def create
       resource.send_enrollment_notification
 
       # Create a history for the enrollment
-      History.enrollment(patient: resource, created_by: current_resource_owner.email, comment: 'User enrolled monitoree via API.')
+      History.enrollment(patient: resource, created_by: resource.creator&.email, comment: 'Monitoree enrolled via API.')
```

Comment:
Some of this hasn't changed... but we should consider decoupling this.

If the API moves to a separate container, the `send_enrollment_notification` might not succeed.

Actually, that could be true regardless if there is a separate container.

If the `send_enrollment_notification` fails, the API returns an internal server error. You don't want that to happen, especially if the M2M API is adding a lot of new patients in bulk.

Ideally, the enrollment notification is a separate job that can retry if necessary and the API does not rely on this succeeding.

Reply:
Yeah I also noticed it failed if I didn't have Redis running. Need to think more about what this will look like when moved to a separate container. I was hoping that could be my focus next week. I have a ticket for it and I'll keep this open.

Reply:
Send enrollment notification does a deliver_later, so this gets put into redis/sidekiq. I don't think that really has anything to do with the container it happens to be running in. It's more because redis wasn't running. Probably should be handled more gracefully in that case, but the container part of the equations should be okay.